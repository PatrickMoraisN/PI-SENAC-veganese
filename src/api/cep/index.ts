const CepAPI = {
  async getCep(cep: string) {
    const resp = await fetch(`https://viacep.com.br/ws/${cep}/json/`)
    const JSON = await resp.json()
    return JSON
  },
}

export default CepAPI
