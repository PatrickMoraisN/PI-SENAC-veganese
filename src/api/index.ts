import foodAPI from './products/foods'

export default foodAPI
