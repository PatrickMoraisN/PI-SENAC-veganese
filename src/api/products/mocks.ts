export const foodsMock = [
  {
    id: 1,
    title: 'Salada Colorful',
    description: 'Salada chefe da casa, bem colorida com ingredientes frescos!',
    labels: ['SALADA'],
    src: 'src/assets/colorful.jpg',
    price: '20,00',
  },
  {
    id: 2,
    title: 'Tacos',
    description: '3 deliciosos tacos com carne de jaca e especiarias da casa!',
    src: 'src/assets/tacos.jpg',
    labels: ['MEXICANO', 'PICANTE'],
    price: '22,00',
  },
  {
    id: 3,
    title: 'Expresso grains',
    description: 'Refeição básica perfeita para petiscar com os amigos!',
    src: 'src/assets/grains.jpg',
    labels: ['GRÃOS'],
    price: '15,90',
  },
  {
    id: 4,
    title: 'Expresso Gelado',
    description:
      'Refeição de grãos gelados com uma culinária totalmente nova no Brasil',
    src: 'src/assets/gelado.jpg',
    labels: ['GELADO'],
    price: '29,90',
  },
  {
    id: 5,
    title: 'Salada Green',
    description: 'Meio a meio frutas e folhas com leite vaporizado',
    src: 'src/assets/green.jpg',
    labels: ['SALADA', 'VAPOR'],
    price: '25,90',
  },
  {
    id: 6,
    title: 'Salada Green',
    description: 'Meio a meio frutas e folhas com leite vaporizado',
    src: 'src/assets/green.jpg',
    labels: ['SALADA', 'VAPOR'],
    price: '25,90',
  },
  {
    id: 7,
    title: 'Salada Green',
    description: 'Meio a meio frutas e folhas com leite vaporizado',
    src: 'src/assets/green.jpg',
    labels: ['SALADA', 'VAPOR'],
    price: '25,90',
  },
  {
    id: 8,
    title: 'Salada Green',
    description: 'Meio a meio frutas e folhas com leite vaporizado',
    src: 'src/assets/green.jpg',
    labels: ['SALADA', 'VAPOR'],
    price: '25,90',
  },
  {
    id: 9,
    title: 'Macchiato',
    description: 'O tradicional café feito com água quente e grãos moídos',
    src: 'src/assets/macchiato.png',
    labels: ['SALADA'],
    price: '12,90',
  },
  {
    id: 10,
    title: 'Gurjao',
    description: 'Gurjão de carne de jaca extremamente delicioso e crocante',
    src: 'src/public/static/img/gurjao.jpg',
    labels: ['PETISCOS'],
    price: '39,90',
  },
  {
    id: 11,
    title: 'Gurjao',
    description: 'Gurjão de carne de jaca extremamente delicioso e crocante',
    src: 'src/public/static/img/gurjao.jpg',
    labels: ['PETISCOS'],
    price: '39,90',
  },
  {
    id: 12,
    title: 'Gurjao',
    description: 'Gurjão de carne de jaca extremamente delicioso e crocante',
    src: 'src/public/static/img/gurjao.jpg',
    labels: ['PETISCOS'],
    price: '39,90',
  },
  {
    id: 13,
    title: 'Hamburgão',
    description:
      'Hamburguer delicioso com carne de caju, especialidade da casa!',
    src: 'src/assets/hamburgao.jpg',
    labels: ['ARTESANAL'],
    price: '15,90',
  },
  {
    id: 14,
    title: 'Hamburgão',
    description:
      'Hamburguer delicioso com carne de caju, especialidade da casa!',
    src: 'src/assets/hamburgao.jpg',
    labels: ['ARTESANAL'],
    price: '15,90',
  },
]

export interface Food {
  id: number
  title: string
  description: string
  src: string
  labels: string[]
  price: string
}

export type FoodsListProps = Food[]
