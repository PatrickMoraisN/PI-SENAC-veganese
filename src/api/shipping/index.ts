/* eslint-disable no-var */
/* eslint-disable no-redeclare */
// @ts-ignore

export function calcCrow(
  lat1: number,
  lon1: number,
  lat2: number,
  lon2: number,
) {
  const R = 6371 // km
  const dLat = toRad(lat2 - lat1)
  const dLon = toRad(lon2 - lon1)
  var lat1 = toRad(lat1)
  var lat2 = toRad(lat2)

  const a =
    Math.sin(dLat / 2) * Math.sin(dLat / 2) +
    Math.sin(dLon / 2) * Math.sin(dLon / 2) * Math.cos(lat1) * Math.cos(lat2)
  const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a))
  const d = R * c
  return d
}

function toRad(Value: number) {
  return (Value * Math.PI) / 180
}
