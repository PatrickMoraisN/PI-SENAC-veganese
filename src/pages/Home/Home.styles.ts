import styled from 'styled-components'

export const main = styled.main`
  min-height: 110vh;
`

export const OurCoffees = styled.div`
  max-width: 1120px;
  width: 100%;
  margin: 0 auto;
  h2 {
    padding: 0 32px;
    margin-bottom: 40px;
  }
`

export const IntroContainer = styled.div`
  width: 100%;
  height: 544px;
  background-image: url('src/assets/Background.svg');
`

export const IntroWrapper = styled.div`
  max-width: 1120px;
  width: 100%;
  margin: 0 auto;
  padding: 0 32px;
  width: 100%;
  display: flex;
  gap: 56px;

  img {
    width: 50%;
    margin-top: 94px;
    max-width: 476px;
    max-height: 360px;
  }
`

export const IntroText = styled.div`
  width: 50%;
  height: 192px;
  margin-top: 94px;
  top: 234px;

  div {
    margin-bottom: 66px;
  }
`

export const ItemsContainer = styled.div`
  display: flex;
  width: 100%;
  margin: 0 auto;
  gap: 20px;
  height: 84px;
  flex-wrap: wrap;

  div {
    display: flex;
    margin: 0;
    gap: 8px;

    p {
      flex-wrap: wrap;
      width: 100%;
    }
  }
`

export const CoffeesListWrapper = styled.div`
  max-width: 1120px;
  width: 100%;
  margin: 0 auto;
  padding: 0 32px;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  gap: 32px;
`
