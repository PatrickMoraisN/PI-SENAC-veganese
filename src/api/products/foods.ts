import { FoodsListProps, foodsMock } from './mocks'

const foodAPI = {
  getFoods(): Promise<FoodsListProps> {
    return new Promise((resolve, _reject) => {
      window.setTimeout(() => {
        return resolve(foodsMock)
      }, Math.random() * 2000 + 1000)
    })
  },
}

export default foodAPI
