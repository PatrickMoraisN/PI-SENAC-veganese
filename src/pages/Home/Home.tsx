import React, { useEffect, useState } from 'react'
import { Header } from '@/components/Header'
import Delivery from '@/assets/delivery.svg'
import * as S from './Home.styles'
import foodAPI from '@/api'
import { Card } from './components/Card'
import { FoodsListProps } from '@/api/products/mocks'
import { Coffee, Package, ShoppingCart, Timer } from 'phosphor-react'

export function Home() {
  const [coffeeList, setCoffeeList] = useState<FoodsListProps>([])

  const getFoods = async () => {
    const coffees = await foodAPI.getFoods()
    setCoffeeList(coffees)
  }

  useEffect(() => {
    getFoods()
  }, [])
  return (
    <S.main>
      <Header />
      <S.IntroContainer>
        <S.IntroWrapper>
          <S.IntroText>
            <div>
              <h1>
                Encontre a refeição vegana perfeita para qualquer hora do dia
              </h1>
              <br></br>
              <p>
                Com o Vegane-se você recebe sua comida saudável onde estiver, a
                qualquer hora
              </p>
            </div>
            <S.ItemsContainer>
              <div>
                <ShoppingCart size={17} color="#030303" weight="fill" />
                <p>Compra simples e segura</p>
              </div>

              <div>
                <Package size={17} color="#030303" weight="fill" />
                <p>Embalagem mantém a comida intacta</p>
              </div>

              <div>
                <Timer size={17} color="#030303" weight="fill" />
                <p>Entrega rápida e rastreada</p>
              </div>

              <div>
                <Coffee size={17} color="#030303" weight="fill" />
                <p>A comida chega fresquinha</p>
              </div>
            </S.ItemsContainer>
          </S.IntroText>

          <img src={Delivery} />
        </S.IntroWrapper>
      </S.IntroContainer>

      <S.OurCoffees>
        <h2>Nossas comidas</h2>
      </S.OurCoffees>

      {!!coffeeList.length && (
        <S.CoffeesListWrapper>
          <Card coffeeList={coffeeList} />
        </S.CoffeesListWrapper>
      )}
    </S.main>
  )
}
