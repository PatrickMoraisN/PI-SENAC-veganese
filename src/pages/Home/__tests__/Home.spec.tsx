import { Home } from '../Home'
import foodAPI from '@/api'
import { foodsMock } from '@/api/products'
import { RenderWithContext } from '@/test/util/RenderWithContext'
import { renderWithRouter } from '@/test'

jest.spyOn(foodAPI, 'getFoods').mockResolvedValue(foodsMock)

const HomeComponent = () => (
  <RenderWithContext>
    <Home />
  </RenderWithContext>
)

describe('Home', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('should render correctly', () => {
    const { container } = renderWithRouter(<HomeComponent />)

    expect(container).toMatchSnapshot()
  })
})
