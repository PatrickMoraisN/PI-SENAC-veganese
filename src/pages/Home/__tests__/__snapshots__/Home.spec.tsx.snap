// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<div>
  <main
    class="sc-jSUZER etVWYg"
  >
    <header
      class="sc-bcXHqe GMPFw"
    >
      <a
        aria-current="page"
        class="active"
        data-testid="logo"
        href="/"
      >
        <img
          src=""
          width="50px"
        />
      </a>
      <h1
        class="sc-gswNZR heZsfo"
      >
        Vegane-se
      </h1>
      <div
        class="sc-dkrFOg fezFrT"
      >
        <div
          class="sc-hLBbgP jkdhxo"
        >
          <svg
            fill="#8047F8"
            height="18"
            viewBox="0 0 256 256"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="256"
              width="256"
            />
            <path
              d="M128,16a88.1,88.1,0,0,0-88,88c0,75.3,80,132.2,83.4,134.6a8.3,8.3,0,0,0,9.2,0C136,236.2,216,179.3,216,104A88.1,88.1,0,0,0,128,16Zm0,56a32,32,0,1,1-32,32A32,32,0,0,1,128,72Z"
            />
          </svg>
          SENAC
        </div>
        <a
          class=""
          href="/checkout"
        >
          <button
            class="sc-eDvSVe clgNoH"
            data-testid="cart"
          >
            <svg
              fill="#C47F17"
              height="18"
              viewBox="0 0 256 256"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <path
                d="M223.9,65.4l-12.2,66.9A24,24,0,0,1,188.1,152H72.1l4.4,24H184a24,24,0,1,1-24,24,23.6,23.6,0,0,1,1.4-8H102.6a23.6,23.6,0,0,1,1.4,8,24,24,0,1,1-42.2-15.6L34.1,32H16a8,8,0,0,1,0-16H34.1A16,16,0,0,1,49.8,29.1L54.7,56H216a7.9,7.9,0,0,1,6.1,2.9A7.7,7.7,0,0,1,223.9,65.4Z"
              />
            </svg>
          </button>
        </a>
      </div>
    </header>
    <div
      class="sc-iBYQkv gbiUwc"
    >
      <div
        class="sc-ftTHYK bOyEqK"
      >
        <div
          class="sc-pyfCe kRMHUl"
        >
          <div>
            <h1>
              Encontre a refeição vegana perfeita para qualquer hora do dia
            </h1>
            <br />
            <p>
              Com o Vegane-se você recebe sua comida saudável onde estiver, a qualquer hora
            </p>
          </div>
          <div
            class="sc-jrcTuL cpSgFE"
          >
            <div>
              <svg
                fill="#030303"
                height="17"
                viewBox="0 0 256 256"
                width="17"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="256"
                  width="256"
                />
                <path
                  d="M223.9,65.4l-12.2,66.9A24,24,0,0,1,188.1,152H72.1l4.4,24H184a24,24,0,1,1-24,24,23.6,23.6,0,0,1,1.4-8H102.6a23.6,23.6,0,0,1,1.4,8,24,24,0,1,1-42.2-15.6L34.1,32H16a8,8,0,0,1,0-16H34.1A16,16,0,0,1,49.8,29.1L54.7,56H216a7.9,7.9,0,0,1,6.1,2.9A7.7,7.7,0,0,1,223.9,65.4Z"
                />
              </svg>
              <p>
                Compra simples e segura
              </p>
            </div>
            <div>
              <svg
                fill="#030303"
                height="17"
                viewBox="0 0 256 256"
                width="17"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="256"
                  width="256"
                />
                <path
                  d="M229.9,70.8h0a.1.1,0,0,1-.1-.1,16.2,16.2,0,0,0-6-5.9l-88-49.5a16,16,0,0,0-15.6,0l-88,49.5a16.2,16.2,0,0,0-6,5.9.1.1,0,0,1-.1.1v.2A15,15,0,0,0,24,78.7v98.6a16.1,16.1,0,0,0,8.2,14l88,49.5a16.5,16.5,0,0,0,7.2,2h1.4a15.7,15.7,0,0,0,7-2l88-49.5a16.1,16.1,0,0,0,8.2-14V78.7A15.6,15.6,0,0,0,229.9,70.8ZM128,29.2,207.7,74,177.1,91.4,96.4,46.9Zm.9,89.6L48.4,74,80,56.2l80.8,44.5Zm7.2,103.5.8-89.6L169,114.4v38.1a8,8,0,0,0,16,0V105.3l31-17.6v89.6Z"
                />
              </svg>
              <p>
                Embalagem mantém a comida intacta
              </p>
            </div>
            <div>
              <svg
                fill="#030303"
                height="17"
                viewBox="0 0 256 256"
                width="17"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="256"
                  width="256"
                />
                <path
                  d="M104,16h48a8,8,0,0,0,0-16H104a8,8,0,0,0,0,16Z"
                />
                <path
                  d="M128,32a96,96,0,1,0,96,96A96.2,96.2,0,0,0,128,32Zm45.3,62.1-39.6,39.6a8.2,8.2,0,0,1-11.4,0,8.1,8.1,0,0,1,0-11.4l39.6-39.6a8.1,8.1,0,1,1,11.4,11.4Z"
                />
              </svg>
              <p>
                Entrega rápida e rastreada
              </p>
            </div>
            <div>
              <svg
                fill="#030303"
                height="17"
                viewBox="0 0 256 256"
                width="17"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="256"
                  width="256"
                />
                <path
                  d="M80,56V24a8,8,0,0,1,16,0V56a8,8,0,0,1-16,0Zm40,8a8,8,0,0,0,8-8V24a8,8,0,0,0-16,0V56A8,8,0,0,0,120,64Zm32,0a8,8,0,0,0,8-8V24a8,8,0,0,0-16,0V56A8,8,0,0,0,152,64Zm96,56v8a40,40,0,0,1-37.5,39.9,98,98,0,0,1-27,40.1H208a8,8,0,0,1,0,16H32a8,8,0,0,1,0-16H56.5A96.4,96.4,0,0,1,24,136V88a8,8,0,0,1,8-8H208A40,40,0,0,1,248,120Zm-16,0a24,24,0,0,0-16-22.6V136a92.9,92.9,0,0,1-1.2,15A24,24,0,0,0,232,128Z"
                />
              </svg>
              <p>
                A comida chega fresquinha
              </p>
            </div>
          </div>
        </div>
        <img
          src=""
        />
      </div>
    </div>
    <div
      class="sc-gKPRtg gmirBA"
    >
      <h2>
        Nossas comidas
      </h2>
    </div>
  </main>
</div>
`;
